app/src/main/java/com/codingblocks/cbonlineapp/database/CodeChallengeDao.kt

```diff
@@ -0,0 +1,12 @@
+package com.codingblocks.cbonlineapp.database
+
+import androidx.room.Dao
+import androidx.room.Query
+import com.codingblocks.cbonlineapp.database.models.CodeChallengeModel
+
+@Dao
+interface CodeChallengeDao: BaseDao<CodeChallengeModel> {
+
+    @Query("SElECT * FROM CodeChallengeModel where id = :id")
+    fun getCodeChallengeById(id: String) : CodeChallengeModel
```

Comment:
why this isn't live data?

Reply:
Either make it live data or make it a suspended function otherwise this can lead to a crash in some phones especially Motorola

Reply:
@prathamk22 this one is still left